### PR TITLE
ddl, executor, server: add drop if exists warnings

### LINKED ISSUE #7867: DROP TABLE IF EXISTS behavior difference
## Bug Report

Please answer these questions before submitting your issue. Thanks!

1. What did you do?
If possible, provide a recipe for reproducing the error.

```
tidb> drop table if exists idontexistatall;
Query OK, 0 rows affected (0.00 sec)
```

2. What did you expect to see?

```
mysql57> drop table if exists idontexistatall;
Query OK, 0 rows affected, 1 warning (0.00 sec)

mysql57> show warnings;
+-------+------+--------------------------------------+
| Level | Code | Message                              |
+-------+------+--------------------------------------+
| Note  | 1051 | Unknown table 'test.idontexistatall' |
+-------+------+--------------------------------------+
1 row in set (0.00 sec)
```

3. What did you see instead?

TiDB does not return a warning in the scenario the table didn't exist.

4. What version of TiDB are you using (`tidb-server -V` or run `select tidb_version();` on TiDB)?

```
MySQL [test]> select tidb_version()\G
*************************** 1. row ***************************
tidb_version(): Release Version: v2.1.0-rc.3-25-g3104c8723
Git Commit Hash: 3104c87230600e7514b3a7190689ebea30a1d8ee
Git Branch: newmaster
UTC Build Time: 2018-10-10 02:41:43
GoVersion: go version go1.11 linux/amd64
Race Enabled: false
TiKV Min Version: 2.1.0-alpha.1-ff3dd160846b7d1aed9079c389fc188f7f5ea13e
Check Table Before Drop: false
1 row in set (0.00 sec)
```

This bug does not affect any programs to my knowledge, but is a nice to have compatibility issue.

### PR DESCRIPTION
**Pull request by Rustin-Liu**
Signed-off-by: Rustin-Liu <rustin.liu@gmail.com>

### What problem does this PR solve? 

close #7867
### What is changed and how it works?

Add warnings.
### Check List 

Tests 

 - Unit test

Code changes

None

Side effects

None
Related changes

None

Release note

None

### CODE REVIEW

**Comment by Rustin-Liu** [comment]
@sre-bot /run-all-tests

File: executor/ddl_test.go [hunk 1]
```diff
@@ -1105,3 +1105,13 @@ func (s *testRecoverTable) TestRenameTable(c *C) {
 	tk.MustExec("drop database rename1")
 	tk.MustExec("drop database rename2")
 }
+
+func (s *testRecoverTable) TestDropTableIfExists(c *C) {
```
Comment (bb7133) [inline executor/ddl_test.go]: Please move this test to: https://github.com/pingcap/tidb/blob/1aa0a7d6b304fb3efb1a2fe5689db07b21168d5d/executor/ddl_test.go#L77

, add tests for `VIEW` here:
https://github.com/pingcap/tidb/blob/1aa0a7d6b304fb3efb1a2fe5689db07b21168d5d/executor/ddl_test.go#L178

and add tests for `SEQUENCE` here:
https://github.com/pingcap/tidb/blob/1aa0a7d6b304fb3efb1a2fe5689db07b21168d5d/ddl/sequence_test.go#L90

BTW, as you may noticed, the testing codes for `DDL` is a mess, we have a plan to re-organize them.

Comment (Rustin-Liu) [reply executor/ddl_test.go]: All fixed.

**Comment by Rustin-Liu** [comment]
@sre-bot  /run-all-tests

File: executor/ddl_test.go [hunk 2]
```diff
@@ -173,6 +173,13 @@ func (s *testSuite6) TestCreateTable(c *C) {
 	tk.MustExec("insert into create_auto_increment_test (name) values ('aa')")
 	r = tk.MustQuery("select * from create_auto_increment_test;")
 	r.Check(testkit.Rows("1000 aa"))
+
+	// test for drop if exists.
```
Comment (zimulala) [inline executor/ddl_test.go]:
```suggestion
	// Test for `drop table if exists`.
```

File: ddl/sequence_test.go [hunk 3]
```diff
@@ -152,6 +152,10 @@ func (s *testSequenceSuite) TestDropSequence(c *C) {
 	_, err = tk1.Exec("drop sequence my_seq")
 	c.Assert(err, NotNil)
 	c.Assert(err.Error(), Equals, "[planner:1142]DROP command denied to user 'localhost'@'myuser' for table 'my_seq'")
+
+	// test for if exists.
```
Comment (zimulala) [inline ddl/sequence_test.go]:
```suggestion
	// Test for `drop sequence if exists`.
```

Comment (Rustin-Liu) [reply ddl/sequence_test.go]: refactored.

**Comment by tangenta** [comment]
@Rustin-Liu The changes in `server/http_handler_test.go` and `server/server_test.go` seem to be irrelavent.

**Comment by Rustin-Liu** [comment]
> @Rustin-Liu The changes in `server/http_handler_test.go` and `server/server_test.go` seem to be irrelavent.

Could you please explain why can not do this change?

**Comment by tangenta** [comment]
> > @Rustin-Liu The changes in `server/http_handler_test.go` and `server/server_test.go` seem to be irrelavent.
> 
> Could you please explain why we can not do this change?

I'm just confused by this code.

**Comment by Rustin-Liu** [comment]
@tangenta  Because if the Strict if true, we will get err.
https://github.com/pingcap/tidb/blob/7e791709bdb7534d009127d2061f3916e5ebb6d1/server/http_handler_test.go#L941

**Comment by zimulala** [comment]
@AilinKid is dealing with this problem.

**Comment by Rustin-Liu** [comment]
@sre-bot  /merge

**Comment by Rustin-Liu** [comment]
@sre-bot /run-auto-merge

**Comment by Rustin-Liu** [comment]
@sre-bot  You are stupid🙄

**Comment by AilinKid** [comment]
> @sre-bot You are stupid🙄

don't worry about it, it should merge the tidb-test PR first then to it.

**Comment by Rustin-Liu** [comment]
> > @sre-bot You are stupid🙄
> 
> don't worry about it, it should merge the tidb-test PR first then to it.

Got it.  Thanks for your supporting!

**Pull request merged**